tests/test_cloak_shield_delay.c

```
@@ -0,0 +1,169 @@
+
+    /* Shield HP must be PRESERVED, not zeroed, through the transition. */
+    for (int i = 0; i < BC_MAX_SHIELD_FACINGS; i++)
+        ASSERT(ship.shield_hp[i] == cls->shield_hp[i]);
+}
+
+/*
+ * #192-C: Decloak grace window.
+ * After the decloak transition completes, the ship is fully visible but shields
+ * stay INACTIVE for ~1.0s, then re-enable.
+ */
+TEST(decloak_grace_window)
```

Clang (14.0.6)

[warning] function 'test_decloak_grace_window' has cognitive complexity of 27 (threshold 25)

(readability-function-cognitive-complexity)